app/components/Connect4/lib/matches/diagonal.js

```diff
@@ -38,36 +38,29 @@ function isTopRight(grid) {
 
     found = 0;
     foundPiece = 0;
-    col = baseCol;
+    col = baseCol - 1; // Subtracting 1 to compensate for incrementing col at
+                       // the beginning of the loop
 
-    // Here we work our way up the current diagonal
+    // Here we work our way *UP* the current diagonal
     for (let row = 0; row < numRows; row++) {
+      col++;
 
       // Ensure that the given column and row are on the board
       if (col >= 0 && col < numCols && row < numRows) {
 
         let piece = grid[col][row];
 
-        // Does nothing if piece is zero
-        if (!piece) {
-          continue;
+        if(!piece) {
+          found = 0;
         }
 
-        if (!foundPiece) {
-          foundPiece = piece;
```

Comment:
Even if there is currently a `foundPiece` (i.e. `!foundPiece == false`), you need to set `foundPiece` to the current piece for correct counting on the next iteration.